provider/source/src/datetime/legacy.rs

```diff
@@ -5,23 +5,15 @@
 //! TODO(#5613): Even though these markers are no longer exported, we need them in order to export
 //! semantic skeleton data markers. This should be refactored to skip the intermediate data struct.
 
+#![allow(dead_code)] // so many unused fields
```

Comment:
Issue: I want to eventually get rid of this whole file. If fields are actually dead, delete them?

Reply:
I did do that whole work, but it's super messy because you end up with structs with single or even no fields, and I think it makes the tests a lot harder to parse because you lose a lot of semantics of what these are. So I reverted to this.

Reply:
let's keep this massive cleanup for a different PR please

Reply:
These blanket allows make future improvements harder because we won't notice when we actually can delete code

Reply:
> A file containing code we actually want to delete is like the worst file to put a blanket suppression

I disagree: if we want to delete the code here we should have a followup issue whose job it is to track the cleanup. I don't think other PRs should be on the hook for that.